doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1671 @@
+Remember, any traffic sent by the host may end up on the other homing TOR based on the bundle-Ethernet interface hashing. This is where MAC, ARP and ND table synchronization come to play. All learned entries (local adjacencies) on a TOR are synchronized to all other peering TORs.
+2. Inter-subnet communication: Host routing enables the inter-subnet / routing capability of a local host with remote host which are in a different subnet. By advertising a more specific prefix than usual gateway address (SVI-IP), remote TORs know exactly where a host resides on the network. Optimal forwarding can always be achieved.
+FRR has the capability to synchronize MAC FDB and SVI-IP adjacency between peering PEs. ARP/ND adjacency prefixes are programmed in RIB/FIB on remote TOR.
+
+<a id=""></a>
+### 2.2.8 Mutual exclusion: MCLAG and EVPN MH
+The MCLAG configuration will not be allowed if ESI is configured on one or more interface(s).
+
+<a id=""></a>
+### 2.2.9 ARP and ND suppression
+ARP/ND suppression will be supported in EVPN Multihoming scenarios. The VTEP will respond to the ARP/ND requests received on local access ports only for the ARP/ND installed against remote VTEPs. ARP/ND response will not be generated by the VTEP if ARP/ND is synchronized and installed on the local Multihomed ES, even though the ARP/ND learning happened on the remote VTEP multihoming a given ES.
+
+<a id="3-Design"></a>
+# 3 Design
+
+<a id="31-Overview"></a>
+## 3.1 Overview
+
+<a id="32-DB-Changes"></a>
+## 3.2 DB Changes
+
+<a id="321-CONFIG-DB"></a>
+### 3.2.1 CONFIG DB
+
+**EVPN_ETHERNET_SEGMENT**
+
+```
+;New table
+;Specifies EVPN Ethernet Segment interface
+;
+; Status: stable
+key = EVPN_ETHERNET_SEGMENT|"ifname"if_id
+; field = value
+esi        = "AUTO"  or es_id
+                ; es_id is 10 byte colon (:) separated string when type = "TYPE_0_OPERATOR_CONFIGURED".
+                ; Otherwise, esi value should be "AUTO" falling back on ESI type 1 or 3.
+type       = esi_type
+                ; esi_type should be string with one of the below values:
+                ; "TYPE_0_OPERATOR_CONFIGURED" for Type-0 ESI
+                ; "TYPE_1_LACP_BASED" for Type-1 ESI
+                ; "TYPE_3_MAC_BASED" for Type-3 ESI
+ifname     = "ifname"if_id
+                ; if_id is the interface identifier. Same value as in the key.
+                ; e.g., PortChannel1, where the ifname is PortChannel and the if_id is 1.
+df_pref    = 1*5DIGIT
+                ; Designated-Forwarder election preference for this router in (1..65535) range.
+                ; Default=32767.
+```
```

Comment:
EVPN_ETHERNET_SEGMENT table does not map on existing FRR commands and is missing required functionality: mac and id for type-3 ESI. Those 2 items are both mandatory in FRR for type-3 ES to work. Also to our knowledge there is no type-1 ESI support in FRR right now or in any PR. So even if everything else is ready to resolve ESI through LACP, FRR is not ready at all.

Reply:
We dont care about ESI value in SONIC. FRR using the Ethernet Segment info programs needed DB in SONIC such as SHL, DF/NDF, etc.

Reply:
@AntonButenkoGL , you are right that FRR does not support Type-1 ESI at present. Type-1 ESI requires enhancement of FRR by passing the lacp partner mac through fpm. Type-1 ESI support can be brought into SONiC once FRR has it.

I have removed the mention of type-1 ESI from this hld in the next commit. However, this comment in the db description will remain as is. This convention of specifying AUTO for esi is borrowed from ocyang:
https://github.com/openconfig/public/blob/master/doc/evpn_use_cases.md#ethernet-segment